### PR TITLE
Make all miner settings configurable at runtime

### PR DESCRIPTION
**Pull request by jcnelson**
This PR backports #4061 from `develop` to `master` and further makes every setting in `[miner]` and most miner-related settings in `[burnchain]` configurable at runtime. The config file will be reloaded whenever the relevant fields are accessed.

### CODE REVIEW

File: testnet/stacks-node/src/config.rs [hunk 1]
```diff
@@ -394,6 +398,28 @@ lazy_static! {
...
+    /// get the up-to-date miner options from the config
+    pub fn get_miner_config(&self) -> Result<MinerConfig, String> {
+        if let Some(path) = &self.config_path {
+            let config_file = ConfigFile::from_path(path.as_str())?;
+            let config = Config::from_config_file(config_file)?;
+            Ok(config.miner)
+        } else {
+            Ok(self.miner.clone())
+        }
+    }
```
Comment (kantai) [inline testnet/stacks-node/src/config.rs:421]: Looking at the invocations of `get_miner_config` (and, for that matter `get_burnchain_config`), it seems like every caller does the same defaulting behavior, so I think it makes sense to just do:

```suggestion
    /// re-read an up-to-date config from this config file's path, or if there is no path,
    /// just return a clone of the current config.
    fn reload_config(&self) -> Result<Config, String> {
        let Some(path) = &self.config_path else {
            return self.clone()
        };
        let config_file = ConfigFile::from_path(path.as_str())?;
        Config::from_config_file(config_file)
    }
    
    /// re-read the up-to-date miner options from the config file's path, or if there is 
    /// no path or it is unreadable, just return the current miner config.
    pub fn get_miner_config(&self) -> MinerConfig {
        let config = self.reload_config().unwrap_or_else(|e| {
             warn!("Failed to reload miner config: {e}");
             self.clone()
        });
        config.miner
    }
```

**Review by kantai — COMMENTED** [review]
LGTM, just had a comment about some opportunities to reduce repetition.

But before merging into `master`, this needs a changelog entry.

**Review by obycode — APPROVED** [review]
lgtm!

**Comment by jcnelson** [comment]
Oh fuck me `cargo fmt` just blew this diff up.  Lemme revert

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to the versioning scheme outlined in the [README.md](README.md).
 
+## [Unreleased]
```
Comment (kantai) [inline CHANGELOG.md:8]: This should be `[2.4.0.0.5]` on merging to `master`

**Comment by jcnelson** [comment]
This is superseded by #4262

**Pull request closed without merging**